Catappult/appcoins-wallet-android
# [feature/APPC-2465] Bottom Navigation - Home

**What does this PR do?**

- Refactors home from java to kotlin -> from TransactionsActrivity.java to HomeFragment.kt.
- Implements new BottomNavigation from HomeFragment and every navigation. 
- Refactors home view model from java to kotlin -> from TransactionViewModel.java to HomeViewModel.kt
- Implements Single State in HomeViewModel
- Removes TransactionViewInteractor and changes to UseCases
- Changes a few dependencies to UseCases.


**Database changed?**

No

**Where should the reviewer start?**

- HomeFragment.kt
- HomeViewModel.kt
- HomeNavigator.kt
- UseCaseModule.kt

**How should this be manually tested?**

- Open Wallet
- Test Promotions bottom navigation tooltip
- Close app, open again and test fingerprint tooltip
- Test navigation 
   - to promotions (both from the empty screen coins and from the BottomNavigation)
   - to TopUp
   - to balance (both from the balance value string and from the BottomNavigation). ps: MyWallets not implemented, so should show a blank screen.
   - to send
   - to receive
   - to Support
   - to Settings
- In settings, recover a wallet and check navigation back to home fragment.
- Check the balance values
- Check the transactions list
- navigate to transaction details
- Check navigation back to home and performance (list doesn't need to refresh)
- Go to settings to change wallets (recover again) 
- Check the balance values and transactions list

**What are the relevant tickets?**

 https://aptoide.atlassian.net/browse/APPC-2465


**Code Review Checklist**
